app/src/renderer/vuex/modules/validators.js

```diff
@@ -54,6 +60,24 @@ export default ({ node }) => {
       if (validatorHash === state.validatorHash) return
       commit(`setValidatorHash`, validatorHash)
       await dispatch(`getValidators`)
+    },
+    async getValidatorDistributionInfoAndRewards({ commit }, validatorAddr) {
+      state.loading = true
+      try {
+        let { self_bond_rewards, val_commission } = await node.getValidatorDistributionInformation(validatorAddr)
+        const rewardsArray = await node.getValidatorRewards(validatorAddr)
+        const rewards = coinsToObject(rewardsArray)
+        self_bond_rewards = coinsToObject(self_bond_rewards)
+        val_commission = coinsToObject(val_commission)
+        const info = { self_bond_rewards, val_commission, rewards }
+        commit(`setValidatorDistributionInfo`, { validatorAddr, info })
+        state.error = null
+      } catch (error) {
+        Sentry.captureException(error)
+        state.error = error
+      }
+      state.loading = false
+      state.loaded = true
```

Comment:
`state.loaded = true` should only happen inside the try, not if there is an error (then it would be false).

Reply:
👍 updated